plugin-server/functional_tests/session-recordings.test.ts

```diff
@@ -140,6 +168,67 @@ test.concurrent(
     20000
 )
 
+test.concurrent(`recording events not ingested to ClickHouse if team is opted out`, async () => {
+    // NOTE: to have something we can assert on in the positive to ensure that
+    // we had tried to ingest the recording for the team with the opted out
+    // session recording status, we create a team that is opted in and then
+    // ingest a recording for that team. We then create a team that is opted in
+    // and ingest a recording for that team. We then assert that the recording
+    // for the team that is opted in was ingested and the recording for the team
+    // that is opted out was not ingested.
+    const tokenOptedOut = uuidv4()
+    const teamOptedInOut = await createTeam(postgres, organizationId, undefined, tokenOptedOut, false)
+    const uuidOptedOut = new UUIDT().toString()
+
+    await capture(
+        producer,
+        null,
+        new UUIDT().toString(),
+        uuidOptedOut,
+        '$snapshot',
+        {
+            $session_id: '1234abc',
+            $snapshot_data: 'yes way',
+        },
+        tokenOptedOut,
+        new Date(),
+        new Date(),
+        new Date(),
+        'session_recording_events'
+    )
+
+    const tokenOptedIn = uuidv4()
+    const teamOptedInId = await createTeam(postgres, organizationId, undefined, tokenOptedIn)
+    const uuidOptedIn = new UUIDT().toString()
+
+    await capture(
+        producer,
+        null,
+        new UUIDT().toString(),
+        uuidOptedIn,
+        '$snapshot',
+        {
+            $session_id: '1234abc',
+            $snapshot_data: 'yes way',
+        },
+        tokenOptedIn,
+        new Date(),
+        new Date(),
+        new Date(),
+        'session_recording_events'
+    )
+
+    await waitForExpect(async () => {
+        const events = await fetchSessionRecordingsEvents(clickHouseClient, teamOptedInId)
+        expect(events.length).toBe(1)
+    })
+
+    // NOTE: we're assuming that we have a single partition for the Kafka topic,
+    // and that the consumer produces messages in the order they are consumed.
```

Comment:
had exactly this question - do we have a single partition here? How is this confirmed?

should we say in the comment that otherwise the test can be flaky if the code is broken or is that part obvious?

Reply:
Have added a comment for now.

Reply:
When we're dropping session recordings that aren't enabled we should somehow track that - be it ingestion warning or prometheus metric too. 

But regardless of that we should check that it isn't going to be stored - is there a way for us to force for this test for there to be only a single kafka partition?